### PR TITLE
[unittest/models] Fix matching for models unittest

### PR DESCRIPTION
**Pull request by kparichay**
This patch fixes the matching for the last layer of each model for the
unittest of the models.

See also #1562

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: test/unittest/unittest_nntrainer_models.cpp [hunk 1]
```diff
@@ -423,18 +429,23 @@ void GraphWatcher::compareFor(const std::string &reference,
...
       nn.backwarding(label, iteration);
 
-      for (auto it = nodes.rbegin(); it != nodes.rend() - 1; it++) {
-        if (it == nodes.rend())
-          it->backward(iteration, true, !optimize);
-        else
+      for (auto it = nodes.rbegin(); it != nodes.rend(); it++) {
+        if (it == nodes.rend() - 1) {
+          /** check last layer backwarding only when not input layers */
+          if (it->getNodeType() != nntrainer::InputLayer::type)
```
Comment (zhoonit) [inline test/unittest/unittest_nntrainer_models.cpp:444]: Could you change this to query `getNumInputConnections()`?

File: test/unittest/unittest_nntrainer_models.cpp [hunk 2]
```diff
@@ -423,18 +429,23 @@ void GraphWatcher::compareFor(const std::string &reference,
                   nntrainer::Tensor::epsilon);
     }
 
-    for (auto it = nodes.begin(); it != nodes.end() - 1; ++it) {
-      it->forward(iteration, *(it + 1));
+    auto it = nodes.begin();
+    for (; it != nodes.end() - 1; ++it) {
+      it->forward(iteration, !(it + 1)->supportInPlace());
     }
+    it->forward(iteration, true);
```
Comment (zhoonit) [inline test/unittest/unittest_nntrainer_models.cpp:436]: Nice catch :)

**Review by lhs8928 — APPROVED** [review]
LGTM

File: test/unittest/unittest_nntrainer_models.cpp (hunk 1 above)
Comment (kparichay) [reply test/unittest/unittest_nntrainer_models.cpp:444]: This is intentional. First layer can be convolution but we want to verify backwarding for it.
Updated rather to supportBackwarding.

Comment (zhoonit) [reply test/unittest/unittest_nntrainer_models.cpp]: I see :)

**Review by zhoonit — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**